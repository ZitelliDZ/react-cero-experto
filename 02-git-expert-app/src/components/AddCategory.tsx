import { useState } from "react";

interface AddCategoryProps {
  handleAdd: (category: string) => void;
}
export const AddCategory = ({ handleAdd }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState("Hola Mundo");

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (inputValue.trim().length < 1) return;

    handleAdd(inputValue.trim());
    setInputValue("");
  };

  return (
    <form action="" onSubmit={(event) => handleSubmit(event)} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs..."
        value={inputValue}
        onChange={(event) => handleInputChange(event)}
      />
    </form>
  );
};


