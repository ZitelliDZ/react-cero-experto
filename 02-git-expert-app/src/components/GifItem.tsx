
interface GifItemProps {
    url: string;
    title: string;
    

}

export const GifItem = ({ url,title }: GifItemProps) => {
  return (
    
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


