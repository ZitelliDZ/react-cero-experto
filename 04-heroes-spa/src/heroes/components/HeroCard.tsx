import { Link } from "react-router-dom";
import { Hero } from "../data/heroes";

interface HeroCardProps {
    hero: Hero;
    buttonMas?:boolean;
    }

export const HeroCard = ({hero,buttonMas=false}: HeroCardProps) => {
  return (
    <div className="d-flex w-auto animate__animated animate__fadeInLeft">
        
    <div className="card" style={{width: '18rem'}}>
        <img src={`/assets/heroes/${hero.id}.jpg`} className="card-img-top " alt={hero.superhero}/>
        <div className="card-body">
            <h5 className="card-title">{hero.superhero}</h5>
            <p className="card-text">{hero.alter_ego}</p>
            {
                (hero.alter_ego !== hero.characters)
                && <p className="card-text">{hero.characters}</p>
            }
            <p className="card-text">
                <small className="text-muted">{hero.first_appearance}</small>
            </p>
            {
                buttonMas && 
                <Link to={`/hero/${hero.id}`} className="btn btn-primary">MÃ¡s...</Link>
            }
        </div>
      
    </div>
    </div>
  )
}
