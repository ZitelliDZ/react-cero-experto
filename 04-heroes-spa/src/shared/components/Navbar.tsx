import { useContext } from "react";
import { NavLink, useNavigate } from "react-router-dom";
import { AuthContext } from "../../auth/context/AuthContext";

export const Navbar = () => {

  const navigate = useNavigate();
  const {user, authState, logout } = useContext(AuthContext);


  const onLogout = () => {
    logout();
    navigate("/login", { replace: true });
  };
  return (
    <nav className="navbar navbar-expand-lg   text-white bg-dark">
      <div className="container">
      <div className="container-fluid ">
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li>
              <NavLink 
                to="/"
                className={(args) =>
                  `nav-link  text-white ${args.isActive ? "active bg-danger" : ""}`
                }
              >
                Home
              </NavLink>
            </li>
            <li>
              <NavLink 
                to="/dc"
                className={(args) =>
                  `nav-link  text-white ${args.isActive ? "active bg-danger" : ""}`
                }
              >
                DC
              </NavLink>
            </li>
            <li>
              <NavLink 
                to="/marvel"
                className={(args) =>
                  `nav-link  text-white ${args.isActive ? "active bg-danger" : ""}`
                }
              >
                Marvel
              </NavLink>
            </li>
            <li>
              <NavLink 
                to="/login"
                className={(args) =>
                  `nav-link  text-white ${args.isActive ? "active bg-danger" : ""}`
                }
              >
                Login
              </NavLink>
            </li> 
            <li>
              <NavLink 
                to="/search"
                className={(args) =>
                  `nav-link  text-white ${args.isActive ? "active bg-danger" : ""}`
                }
              >
                Search
              </NavLink>
            </li>
          </ul>
        </div>
      </div>

      <div className="collapse navbar-collapse   order-3 justify-content-end">
        <div className="navbar-nav ml-auto align-items-center gap-4">
                <span>
                  {user?.name}
                </span>
                <button className="nav-link nav-item btn btn-danger  text-white" onClick={onLogout}>
                  Logout

                </button>
        </div>

      </div>
      </div>
    </nav>
  );
};
