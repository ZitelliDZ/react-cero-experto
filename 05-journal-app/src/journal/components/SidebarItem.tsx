import { TurnedIn, TurnedInNot } from "@mui/icons-material";
import {
  Grid,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from "@mui/material";
import { useMemo } from "react";
import { useDispatch, useSelector } from "react-redux";
import { setActiveNote } from "../../store/journal/journalSlice";

export const SidebarItem = ({ note }: any) => {

    const dispatch = useDispatch(); 
    const { active } = useSelector((state) => state.journal);

    const isActive = active?.id === note.id;

    const newTitle = useMemo(() => {
        return note.title.length > 17 ? note.title.slice(0, 17) + "..." : note.title;
    }, [note.title]);


    const handleActiveNote = () => {
        if(note.id !== active?.id){
            dispatch(setActiveNote(note))
        }
    }
  return (
    <ListItem disablePadding key={note.id}>
      <ListItemButton 
        onClick={handleActiveNote}
        >
        <ListItemIcon>
            { 
                isActive ? <TurnedIn color="primary" /> : <TurnedInNot />
            }
        </ListItemIcon>
        <Grid
          container
          justifyContent="space-between"
          alignItems={"center"}
          direction={"row"}
        >
          <ListItemText primary={newTitle} />
          <ListItemText secondary={note.body} />
        </Grid>
      </ListItemButton>
    </ListItem>
  );
};
