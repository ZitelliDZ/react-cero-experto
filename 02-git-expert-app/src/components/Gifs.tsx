
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from './GifItem';

interface GifsProps {
  category: string;
}

export const Gifs = ({ category }: GifsProps) => {
  const { images, isLoading } = useFetchGifs({ category });

  /*
  const [images, setImages] = useState([]);


  const getImages = async () => {
    const gifs = await getGifs(category);
    setImages(gifs);
  };

  useEffect(() => {
    // getGifs(category).then((imgs) => {
    //   setImages(imgs);
    // });

    getImages();
  }, []);*/

  if ( isLoading ) {
    return <p>Loading...</p>;
  }

  return (
    <div>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img: any) => {
          return <GifItem key={img.id} {...img} />;
        })}
      </div>
    </div>
  );
};


