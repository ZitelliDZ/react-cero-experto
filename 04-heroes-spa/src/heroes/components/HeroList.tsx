import { useEffect, useMemo, useState } from "react";
import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";
import { Hero } from "../data/heroes";
import { HeroCard } from "./HeroCard";

interface HeroListProps {
  publisher: string;
}

export const HeroList = ({ publisher }: HeroListProps) => {
  const [heroes, setHeroes] = useState<Hero[]>([]);


  const herosResult = useMemo(() => getHeroesByPublisher(publisher),[publisher]);
  useEffect(() => {
    try {
      setHeroes(herosResult);
    } catch (error) {
      console.log(error);
    }
  }, [publisher]);

  return (
    <div className="row row-cols-1 row-cols-md-3 g-4">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} hero={hero} buttonMas  />
      ))}
    </div>
  );
};
