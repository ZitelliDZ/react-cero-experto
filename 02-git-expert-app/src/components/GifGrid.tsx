import { Gifs } from "./Gifs";

 

interface GifGridProps {
    categories: string[];
}

export const GifGrid = ({ categories }: GifGridProps) => {


  return (
    <ol>
      {categories.map((category, index) => {
        return <Gifs key={index} category={category} />;
      })}
    </ol>
  );
};


