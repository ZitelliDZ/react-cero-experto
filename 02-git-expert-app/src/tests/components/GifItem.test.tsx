import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components"; 



describe('Pruebas en <GifItem />', () => {

    const title = 'Un titulo';
    const url = 'https://one-punch.com/';
     

    test('debe de hacer match con el snapshot', () => {
        const {container } = render(<GifItem  title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el url y el alt indicados', () => {
        const {  } = render(<GifItem  title={title} url={url} />);
        // screen.debug();
        expect((screen.getByRole('img') as HTMLImageElement).src).toBe(url);
        expect((screen.getByRole('img') as HTMLImageElement).alt).toBe(title);


        const { src, alt } = (screen.getByRole('img') as HTMLImageElement);
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem  title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
 

});
