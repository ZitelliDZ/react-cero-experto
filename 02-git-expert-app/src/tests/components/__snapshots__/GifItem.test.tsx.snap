// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://one-punch.com/"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
